### PR TITLE
add getCommentsByID controller and model, plus test case to check ret…

### PR DESCRIPTION
**Pull request by dancancode85**
…urns correct array object

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -115,3 +115,47 @@ describe('getArticles', () => {
...
 });
 
+describe('getCommentsByID', () => {
+it('should correctly return array of objects with a status 200 code', () => {
+  return request(app)
+  .get('/api/articles/1/comments')
+  .expect(200)
+  .then((res) => {
+    const {comments} = res.body
+    console.log(res.body);
```
Comment (aloush-dev) [inline __tests__/integration.test.js]: remember to remove console logs

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -115,3 +115,47 @@ describe('getArticles', () => {
...
+    comments.forEach((comment) => {
+      expect(typeof comment.comment_id).toBe('number');
+      expect(typeof comment.votes).toBe('number');
+      expect(typeof comment.created_at).toBe('string');
+      expect(typeof comment.author).toBe('string');
+      expect(typeof comment.body).toBe('string');
+      expect(typeof comment.article_id).toBe('number');
+    });
```
Comment (aloush-dev) [inline __tests__/integration.test.js:132]: Before we forEach we should check the length of the array to assure it's not empty!

File: models/models.js [hunk 3]
```diff
@@ -60,9 +60,28 @@ const getAllArticles = () => {
     });
 };
 
+const getCommentsByArticleID = (id) => {
+  if (isNaN(id)) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
```
Comment (aloush-dev) [inline models/models.js]: PSQL should handle this error for you instead of a custom promise reject.

File: models/models.js [hunk 4]
```diff
@@ -60,9 +60,28 @@ const getAllArticles = () => {
...
+  return db
+  .query(
+  `SELECT *  FROM comments
+  WHERE article_id = $1 ORDER BY created_at DESC;`,
+  [id]
+  )
+  .then((data) => {
+    if (!data.rows.length || data.rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Not found" });
+    }
```
Comment (aloush-dev) [inline models/models.js]: This isn't right we should return the empty array as it a valid request.

**Review by aloush-dev — CHANGES_REQUESTED** [review]
What happens if we give it a valid ID but the article has no comments? I'd like to see a test for this. It should be a 200 success.


Please resubmit once this test has been added.
Thanks

**Review by aloush-dev — CHANGES_REQUESTED** [review]
What happens if we give it a valid ID but the article has no comments? I'd like to see a test for this. It should be a 200 success.


Please resubmit once this test has been added.
Thanks

File: __tests__/integration.test.js [hunk 5]
```diff
@@ -132,20 +132,20 @@ it('should correctly return array of objects with a status 200 code', () => {
     });
   })
 });
-it('should return 404 when passed an article id that does not exist', () => {
```
Comment (saimakhanom) [inline __tests__/integration.test.js]: this test was correct as it was before - Ali meant if the article id exists but it doesnt actually have any comments - we want a new test that tests this scenario (we dont want to give an error if the article doesnt have any comments, we should just return an empty array

File: errors.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
 exports.handlePSQLErrors = (err, req, res, next) => {
+  console.log(err);
```
Comment (saimakhanom) [inline errors.js:2]: remove logs :)

**Review by Hard-Luck — APPROVED** [review]
changes well made you can merge 👍

**Pull request merged**